packages/react-core/src/components/TextInput/TextInput.tsx

```diff
@@ -218,10 +216,11 @@ export class TextInputBase extends React.Component<TextInputProps, TextInputStat
         aria-invalid={props['aria-invalid'] ? props['aria-invalid'] : validated === ValidatedOptions.error}
         required={isRequired}
         disabled={isDisabled}
-        readOnly={!!readOnlyVariant || isReadOnly || readOnly}
+        readOnly={!!readOnlyVariant || isReadOnly}
         ref={innerRef || this.inputRef}
         {...((customIconUrl || customIconDimensions) && { style: customIconStyle })}
         {...getOUIAProps(TextInput.displayName, ouiaId !== undefined ? ouiaId : this.state.ouiaStateId, ouiaSafe)}
+        {...props}
```

Comment:
I actually leave line 219 as is and not change how we are spreading the props right now.  We could inadvertently break a consumers component.